app/lib/katello/event_daemon/monitor.rb

```diff
@@ -0,0 +1,47 @@
+module Katello
+  module EventDaemon
+    class Monitor
+      def initialize(services)
+        @services = services
+        @service_statuses = {}
+      end
+
+      def start
+        loop do
+          Rails.application.executor.wrap do
+            check_services
+          end
+          sleep 15
+        end
+      end
+
+      def check_services
+        @services.each do |service_name, service_class|
+          @service_statuses[service_name] = service_class.status
+        rescue => error
+          Rails.logger.error("Error occurred while pinging #{service_class}")
+          Rails.logger.error(error.message)
+          Rails.logger.error(error.backtrace.join("\n"))
+        ensure
+          if error || !@service_statuses[service_name]&.dig(:running)
+            begin
+              service_class.close
+              service_class.run
+              @service_statuses[service_name] = service_class.status
+            rescue => error
+              Rails.logger.error("Error occurred while starting #{service_class}")
+              Rails.logger.error(error.message)
+              Rails.logger.error(error.backtrace.join("\n"))
+            ensure
+              error = nil
+            end
+          end
+          Rails.cache.write(
```

Comment:
It's probably safe + preferable to write the cache just once per loop of `check_services` (this is writing on for each service which I realize is how it was before, essentially)